### ISSUE TITLE
Adding ref of related files (idea)

### ISSUE THREAD

**Issue by luffah** [soywod/unfog#42]
Searching for a good tool for managing task.

This tools is really interesting and appear to be really clean to use.
(working currently with timewarrior and taskwarrior  + a lot of TODO/FIXME in files)

I got only an idea of enhancement  : a field to link to external refs (local wiki, code, issue tracker...) and maybe something to add a task directly from this ref (source code or documentation in vim, ... for issue tracker case it would require some kind of server ... ).
Actually, i'm not sure how much it is a good idea, but doing a simple `gf`  (to start a correction) from a task manager would be pleasant.

**Comment by soywod** [comment]
I see your point, but I have no idea how to integrate this to the actual workflow. Any suggestion? Maybe sth like `unfog attachment <id>` could open the default text editor?

**Comment by luffah** [comment]
oh ... i took the wrong repo to declare the issue (originally then idea was for vim).

On Vim, just having a way to access the list item by item (i.e. a buffer normal mode) will allow to do a `gf` (or a link the shortcut to open the `<cfile>` in a new tab — OR open the list of item as with something similar to nerd tree — OR just using the quickfix list).

In fact, it would be interesting to mention attachment and a position in the attachment (with text files it would be line number; in a webpage, this is already contained in url; in a doc a page number maybe).

An experimental usecase :
```
# adding a task for a project linked to 
#  a document file to edit; the notes that could help; a directory with some assets
#  and the a session file for vim that shall be openned with vim
# (it could be url and images too....)
# assume -a is the option to add attachment
$ unfog add Referencing similar tools -p Zim -a /work/git/zim/README.md:40"[gvim %1]",~/Notebooks/Zim/similar_tools.txt,~/Notebooks/Zim/similar_tools/,~/.vim/sessions/zim.vim
# HERE a dirty idea : managing a dedicated folder and its metadata
# unfog could create <unfog-datadir>/<hash or id>/ and add all attachements
# (excluding url; that would require a script to be openned) as symbolic links; 
# and a small db file that contains metadata (the list of file as entered) to store advanced things (line numbers or commands used to open)

$ unfog attachment Zim:referencing   # i don't know if it can find the note like that // just for the example
# list items and propose to open with `open` or to see all as a (logical) dir
1 - /work/git/zim/README.md:40  [ open with `gvim %1`]
2 - ~/Notebooks/Zim/similar_tools.txt
3 - ~/.vim/sessions/zim.vim
Which to open (number or 'd' for openning as a directory)? 1
[ run `gvim /work/git/zim/README.md:40`]

Which to open (number or 'd' for openning as a directory)? 2
[ run `vim /work/git/zim/README.md`]

Which to open (number or 'd' for openning as a directory)? d
[ make links (if not already made); cd and run `nautilus .`; given the directory is not readonly it means to check metadata are updated given the content of the directory // the user could add directly a file here ]

```

**Comment by soywod** [comment]
> oh ... i took the wrong repo to declare the issue (originally then idea was for vim).

No it's fine, since the Vim plugin is just a view. All the logic needs to be done CLI side.

I like your proposal, but it's too "complex" for a first shot. We should start with a smaller and simpler version of this. I will think about it and let you know !

**Comment by soywod** [comment]
I propose to attach a simple text file for now, so the user can put whatever he wants. Then we can try to add more and more logic to open links, load Vim sessions etc. What do you think? In fact it can be seen as an extended description. It looks a bit like what @mg979 proposed on Unfog's predecessor: https://github.com/soywod/kronos.vim/issues/6.

**Comment by luffah** [comment]
Yes, any textarea is enough to feed a script that would give access to documents.
